apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1.Tests/RequestIdTests.cs

```diff
@@ -0,0 +1,215 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Grpc.Core;
+using Google.Api.Gax.Grpc;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+using System.Collections.Generic;
+
+namespace Google.Cloud.Spanner.V1.Tests;
+
+public class RequestIdTests
+{
+    private const string SampleDatabaseName = "projects/proj/instances/inst/databases/db";
+    private const string SampleSessionName = "projects/proj/instances/inst/databases/db/sessions/sess";
+
+    [Fact]
+    public void RequestId_Format()
+    {
+        var invoker = new RetryFakeCallInvoker();
+        var client = new SpannerClientBuilder { CallInvoker = invoker }.Build();
+
+        client.CreateSession(new CreateSessionRequest { Database = SampleDatabaseName });
+
+        // The expected format is 6 parts that break down as such:
+        // {VersionId}.{s_processId}.{ClientId}.{ChannelId}.{RequestSequence}.{AttemptNum}
+        Metadata headerMetadata = invoker.CapturedMetadata[0];
+        // We can't access RequestId.HeaderKey anymore, so use the string literal.
+        var idString = headerMetadata.Get("x-goog-spanner-request-id").Value;
+
+        var parts = idString.Split('.');
+
+        Assert.Equal(6, parts.Length);
+        var versionId = parts[0];
+        var processId = parts[1];
+        var clientId = parts[2];
+        var channelId = parts[3];
+        var requestSequence = parts[4];
+        var attemptNum = parts[5];
+
+        Assert.Equal("1", versionId);
+        Assert.True(ulong.TryParse(processId, out ulong processIdLong));
+        // We want to make sure the random generation ran. The default value for ulong is 0
+        // and with a very high probability this should not be 0 after generation.
+        Assert.NotEqual(0UL, processIdLong);
+        Assert.True(int.TryParse(clientId, out _));
+        Assert.Equal("1", channelId);
+        Assert.True(int.TryParse(requestSequence, out _));
+        Assert.Equal("1", attemptNum);
+    }
+
+    [Fact]
+    public void RequestIdSource_ProcessId_OverwritingBehavior() =>
+        // The process ID should only ever be set once per process.
+        Assert.Throws<InvalidOperationException>(() =>
+            {
+                // Note in the case of test re-runs within the same process the
+                // "first_override" may cause the exception to be thrown as it was
+                // set prior.
+                SpannerClientImpl.ProcessId = 1UL;
+                SpannerClientImpl.ProcessId = 2UL;
+            });
+
+    // Tests for header injection
+    public static TheoryData<Action<SpannerClient>> SpannerClientActions { get; } = new TheoryData<Action<SpannerClient>>
```

Comment:
You are missing the streaming methods here. We should tests those. It can be on separate tests if its easier.

Also, if you see that we don't use in spanner some of the call invoker methods, maybe we can skip those in the interceptor?